docs/controllers/optimization.md

```diff
@@ -0,0 +1,224 @@
+    "kube-public",
+    "kube-system",
+]
+.into_iter()
+.map(|ns| format!("metadata.namespace!={ns}"))
+.collect::<Vec<_>>()
+.join(",");
+let cfg = watcher::Config::default().fields(&ignoring_system_namespaces);
+```
+
+!!! note "Field Selector Limitations"
+
+    Due to [field-selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) limitations, you cannot filter on arbitrary fields, nor can you do set complements (need to enumerate explicitly).
+
+If you find that field-selectors are too constrictive for your set of objects, the problem can generally be solved using by [explicitly labelling](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) the objects you care about and use label selectors instead:
+
+```rust
+let cfg = Config::default().labels("environment in (production, qa)");
+```
+
+### Watching Metadata Only
+
+Generally, **Kubernetes will return all fields** for an object when doing a watch, so while the main tool at our disposal to reduce networked traffic is to reduce the number of watched objects, there is one trick available:
+
+We can ask Kubernetes to only return [TypeMeta] (`.api_version` + `.kind`) + [ObjectMeta] (`.metadata`) in our watches and list calls using `Api::watch_metadata` or the stream analogue [metadata_watcher]; a drop-in replacement for [watcher]:
+
+```rust
+let cfg = watcher::Config::default().fields(&ignoring_system_namespaces);
+let stream = reflector(writer, metadata_watcher(api, cfg)).applied_objects();
+```
+
+While this will not work in all cases (if you need more data than metadata), it is a big improvement when it is feasible.
+
+Given metadata is generally only responsible from a **fraction of the data** in dense objects - and generally receives even 2x benefits for more sparse objects (see also [[#pruning-fields]]) - swapping to metadata_watchers can **significantly reduce reflector sizes** (memory footprint).
+
+TODO: panel snapshot.
+
+This is also a networked simplification and will also **reduce the amount of networked traffic** the controller is responsible for by commensurate amounts.
+
+TODO: panel snapshot.
+
+### Avoid the List Spike
+
+At time of writing, every permanent watch loop setup against kube-apiserver requires a `list` call (to initialise) followed by a long `watch` call starting at the given `resourceVersion` from `list`. De-syncs could happen, and these would force a re-list to re-initialize (forcing a slew of old objects to be passed through controllers again).
+
+This `list` call has **no enforced limits** on how many objects can be requested in one go and has caused kube to run without limits up until recently. This `list` call has been [showed to cause both IO and memory spikes](https://github.com/kube-rs/kube/issues/1209) and requires a future opt-in in in 0.84 via https://github.com/kube-rs/kube/pull/1211 to watcher configs:
+
+```rust
+let cfg = watcher::Config::default().page_size_limit(50);
+```
```

Comment:
The wording here feels a bit awkward, but I don't have a concrete rewording right now.. :/

Reply:
Agreed, this one was more roughly written than the rest. Have made a smaller change for it in https://github.com/kube-rs/website/pull/35/commits/6d6127cd9c510516bc43a7044f8f73b513be13fd + fixing some obvious typos after.

Feel free to propose other suggestions though!